src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -387,14 +443,23 @@ bool SmokeCoAlarmServer::HandleRemoteSelfTestRequest(CommandHandler * commandObj
         }
         else
         {
-            success = opHandler(endpointId);
-            if (!success)
+            success = SetExpressedState(endpointId, ExpressedStateEnum::kTesting);
+            if (success)
             {
-                status = Status::Failure;
+                success = SetTestInProgress(endpointId, true);
+            }
+
+            if (success)
+            {
+                success = emberAfPluginSmokeCoAlarmSelfTestRequestCommand(endpointId);
+
+                UnsetExpressedState(endpointId, ExpressedStateEnum::kTesting);
```

Comment:
This is wrong.  The expressed state needs to change back _when the test is done_.  Which means the app needs to do it.